#
# doing 'ls -l' on synlink will always show lrwxrwxrwx on Solaris
# fix has been submitted to upstream:
#	https://github.com/proftpd/proftpd/pull/1506
#

--- a/tests/t/lib/ProFTPD/Tests/Commands/LIST.pm
+++ b/tests/t/lib/ProFTPD/Tests/Commands/LIST.pm
@@ -3527,7 +3527,12 @@ sub list_symlink_issue940 {
       }
 
       my $file = 'src.link -> dst.d';
-      my $expected = 'lrwxr-xr-x';
+      my $expected;
+      if ($^O eq 'solaris') {
+        $expected = 'lrwxrwxrwx';
+      } else {
+        $expected = 'lrwxr-xr-x';
+      }
       $self->assert($res->{$file} eq $expected,
         test_msg("Expected '$expected', got '$res->{$file}' for '$file'"));
 
